src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -0,0 +1,483 @@
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+
+public class IntegrationTestUtils {
+
+    /**
+     * loadTestConfigs will process all test yaml files in provided directory.
+     *
+     * @param directory The directory with test files to traverse
+     * @return A List of Tests
+     * @throws FileNotFoundException
+     */
+    public List<Tests> loadTestConfigs(String directory) throws FileNotFoundException {
+        List<Tests> tests = new ArrayList<>();
+        final File folder = new File(directory);
+        if (!folder.exists()) {
+            return new ArrayList<>();
+        }
+        processDirectory(folder, tests);
+        return tests;
+    }
+
+    // processDirectory will traverse the subdirectories of 'folder'.
+    // Useful to group related test files in a directory
+    private void processDirectory(final File folder, List<Tests> tests)
+            throws FileNotFoundException {
+        for (final File fileEntry : Objects.requireNonNull(folder.listFiles())) {
+            if (fileEntry.isDirectory()) {
+                processDirectory(fileEntry, tests);
+            } else {
+                tests.add(processTestFile(fileEntry.getPath()));
+            }
+        }
+    }
+
+    private Tests processTestFile(String filename) throws FileNotFoundException {
+        Yaml yaml = new Yaml(new Constructor(Tests.class));
+        InputStream is = new FileInputStream(filename);
+        return yaml.load(is);
+    }
+
+    /**
+     * runTests will execute the tests that match the pattern. If generate is 'true' then baseline
+     * files will be generated from the test configurations
+     *
+     * @param testLists List of tests to run
+     * @param conn Database connection
+     * @param generate Generate Baseline test files instead of running test
+     * @param pattern Pattern to match on the test description, matching tests will be run
+     * @throws IOException
+     * @throws SQLException
+     * @throws IllegalAccessException
+     * @throws InvocationTargetException
+     */
+    public void runTests(List<Tests> testLists, Connection conn, Boolean generate, String pattern)
+            throws IOException, SQLException, IllegalAccessException, InvocationTargetException {
+        Pattern p = Pattern.compile(pattern);
+        for (Tests tests : testLists) {
+            for (TestEntry testEntry : tests.tests) {
+                if (testEntry.skip_reason != null || !p.matcher(testEntry.description).matches()) {
```

Comment:
This was one of the suggestions to be able to run a test by pattern.  I think the idea was to use the pattern to run different sets of tests such as ExecuteQuery, PrepareExecute, PrepareOnly.  Or for example, if the binary data test takes a long time to run it can be separated and run in its own test function.
I'll change it to be case insensitive.